file_management/include/file_management/file_upload/file_manager.h

```diff
@@ -214,6 +212,34 @@ class FileManager :
     return false;
   }
 
+  /*
+    If the user cfg options for delete_stale_data and a log is over 14 days old from
+    the latest time,it will be deleted. This is because the AWS API for PutLogEvents
+    rejects batches with log events older than 14 days.
+  */
+  void deleteStaleData(){
+    if (stale_data_.empty()) {
```

Comment:
Why doesn't this need to be guarded as well? What happens if something is added at the same time?